src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -56,70 +58,36 @@ public class {{ target_kind_name }} {
                 null
             {% endif %};
         {% endfor %}
-        {% if abstract_data_type == "Value"%}
-            {# Reads `Value` class with a single field `value` #}
-            value = {% if fields[0].value_type | is_built_in_type %}
-                        {% if fields[0].value_type == "bytes[]" %}
-                            reader.newBytes();
-                        {% else %}
-                            reader.{{ fields[0].value_type | camel }}Value();
-                        {% endif %}
-                     {% else %}
-                        {{ fields[0].value_type }}.readFrom(reader);
-                     {% endif %}
-        {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
-            {# Reads `Structure` class with multiple fields based on `field.name` #}
-            reader.stepIn();
-            while (reader.hasNext()) {
-                reader.next();
-                String fieldName = reader.getFieldName();
-                switch(fieldName) {
-                    {% for field in fields %}
-                    case "{{ field.name }}":
-                        {{ field.name | camel }} = {% if field.value_type | is_built_in_type %}
-                                                       {% if field.value_type == "bytes[]" %}
-                                                            reader.newBytes();
-                                                       {% else %}
-                                                            reader.{{ field.value_type | camel }}Value();
-                                                       {% endif %}
+
+        {# Reads `Structure` class with multiple fields based on `field.name` #}
+        reader.stepIn();
+        while (reader.hasNext()) {
+            reader.next();
+            String fieldName = reader.getFieldName();
+            switch(fieldName) {
+                {% for field in fields %}
+                case "{{ field.name }}":
+                    {{ field.name | camel }} = {% if field.value_type | is_built_in_type %}
+                                                   {% if field.value_type == "bytes[]" %}
+                                                        reader.newBytes();
+                                                   {% else %}
+                                                        reader.{{ field.value_type | camel }}Value();
+                                                   {% endif %}
+                                                {% else %}
+                                                    {% if field.value_type is containing("ArrayList") %}
+                                                        {{ util_macros::read_as_sequence(field=field) }}
```

Comment:
(🗺️ PR tour) This statement is modified to read fields that are of type `ArrayList` using util_macros `write_as_sequence`. Previously it was not possible for a field in struct to be of sequence type(e.g. It would be inside some nested type `NestedtypeX`). But with this change it is possible to have fields with sequence type (e.g. `ArrayList<Integer>`).